### PR TITLE
Add `gcompris` educational package

### PR DESCRIPTION
**Pull request by guibou**
###### Motivation for this change

Add `gcompris`, which is a mandatory educational package for kids.

###### Things done

- [ X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X ] Tested execution of all binary files (usually in `./result/bin/`)
- [ X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/games/gcompris/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchurl {
+    url = "http://gcompris.net/download/qt/src/gcompris-qt-${version}.tar.xz";
+    sha256 = "1i5adxnhig849qxwi3c4v7r84q6agx1zxkd69fh4y7lcmq2qiaza";
+  };
+
+  enableParallelBuilding = true;
+
+  cmakeFlags = "-DQML_BOX2D_LIBRARY=${qmlbox2d}/${qtbase.qtQmlPrefix}/Box2D.2.0";
+
+  nativeBuildInputs = [ cmake makeWrapper ];
```
Comment (jtojnar) [inline pkgs/games/gcompris/default.nix]: Please add ninja when using cmake.

File: pkgs/games/gcompris/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+stdenv.mkDerivation rec {
+  version = "0.90";
+  name = "gcompris-${version}";
+
+  src = fetchurl {
+    url = "http://gcompris.net/download/qt/src/gcompris-qt-${version}.tar.xz";
+    sha256 = "1i5adxnhig849qxwi3c4v7r84q6agx1zxkd69fh4y7lcmq2qiaza";
+  };
+
+  enableParallelBuilding = true;
```
Comment (jtojnar) [inline pkgs/games/gcompris/default.nix]: cmake builds in parallel by default.

File: pkgs/games/gcompris/default.nix (hunk 1 above)
Comment (guibou) [reply pkgs/games/gcompris/default.nix]: I was not aware of the runtime detection of `ninja` by `cmake`. Thank you. Fixed.

File: pkgs/games/gcompris/default.nix (hunk 2 above)
Comment (guibou) [reply pkgs/games/gcompris/default.nix]: Explicit is better than implicit no? Do you really want me to remove it?

**Comment by nlewo** [comment]
I've got this error:
```
 /nix/store/c90qjb1n7p7gkj4xf29rj8lx84jbm92h-gcompris-0.90/bin/gcompris-qt 
QCursor: Cannot create bitmap cursor; invalid bitmap(s)
exeCount set to:  1
QQmlApplicationEngine failed to load component
qrc:/gcompris/src/core/main.qml:22 module "QtQuick.Controls" is not installed
qrc:/gcompris/src/core/main.qml:23 module "QtQuick.Window" is not installed
qrc:/gcompris/src/core/main.qml:21 module "QtQuick" is not installed
qrc:/gcompris/src/core/main.qml:22 module "QtQuick.Controls" is not installed
qrc:/gcompris/src/core/main.qml:23 module "QtQuick.Window" is not installed
qrc:/gcompris/src/core/main.qml:21 module "QtQuick" is not installed
qrc:/gcompris/src/core/main.qml:22 module "QtQuick.Controls" is not installed
qrc:/gcompris/src/core/main.qml:23 module "QtQuick.Window" is not installed
qrc:/gcompris/src/core/main.qml:21 module "QtQuick" is not installed

Cannot mix incompatible Qt library (version 0x50901) with this library (version 0x50903)
Aborted
```

Any idea about what is happening?

**Comment by guibou** [comment]
@nlewo try to install the package in your env : `nix-env -i /nix/store/c90qjb1n7p7gkj4xf29rj8lx84jbm92h-gcompris-0.90/`. The nixpkgs manual told so, I don't know why...

**Comment by nlewo** [comment]
@guibou ok, it work with `nix-env`.

File: pkgs/games/gcompris/default.nix (hunk 2 above)
Comment (nlewo) [reply pkgs/games/gcompris/default.nix]: Hmm, there are a lot of other attributes that are implicit:) I also think we can let default values to be applied.

Comment (guibou) [reply pkgs/games/gcompris/default.nix]: Done.

**Comment by guibou** [comment]
I fixed the `enableParallelBuilding` as requested by @jtojnar  and added a `.desktop` file entry. However this PR does not see the changes, I wonder why. I will let a few time to github to see if it refreshs itself and if not, I'll open a new PR.

**Comment by nlewo** [comment]
@guibou looks like the PR has now been updated.
@GrahamcOfBorg build gcompris

**Comment by nlewo** [comment]
@guibou could you either fix the `aarch64` build or restrict the platforms to `x86` only?

**Comment by guibou** [comment]
@nlewo 

Done, set to `[ "i686-linux" "x86_64-linux" ]`.

**Comment by nlewo** [comment]
@GrahamcOfBorg build qmlbox2d

**Comment by nlewo** [comment]
Thanks... from my child:)

**Pull request merged**